src/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -88,6 +88,34 @@ public static void EnsureDirectoryExists(string directoryPath)
             }
         }
 
+        public static bool TryEnsureDirectoryExists(string directoryPath)
+        {
+            try
+            {
+                EnsureDirectoryExists(directoryPath);
+
+                return true;
+            }
+            catch
```

Comment:
This looks like a valid scenario: https://github.com/dotnet/cli/commit/333fbbd16770e34276c0f54cec2521d1060fb7d7
@piotrpMSFT - is this still valid or can we remove try catch